File: src/Client/src/app/api/models/meeting-dto.ts
/* tslint:disable */
/* eslint-disable */
import { BookDto } from '../models/book-dto';
import { BookRecommendationForMeetingDto } from '../models/book-recommendation-for-meeting-dto';
import { BookVoteDto } from '../models/book-vote-dto';
import { MeetingUserStateDto } from '../models/meeting-user-state-dto';
export interface MeetingDto {
  bookRecommendations: Array<BookRecommendationForMeetingDto>;
  dateTime: string;
  id: string;
  previousMeeting?: MeetingDto | null;
  state?: string | null;
  userStates: Array<MeetingUserStateDto>;
  votes: Array<BookVoteDto>;
  winningBook?: BookDto | null;
}

File: src/Server/WebApi/Dtos/MeetingDto.cs
namespace Cbc.WebApi.Dtos;

using AutoMapper;
using Cbc.WebApi.Interfaces;
using Cbc.WebApi.Models.Entities;

public class MeetingDto : IMapFrom<Meeting>
{
    public Guid Id { get; set; }

    public DateTime DateTime { get; set; }

    public BookDto? WinningBook { get; set; }

    public MeetingDto? PreviousMeeting { get; set; }

    public string? Status { get; set; }

    public List<BookVoteDto> Votes { get; set; } = [];

    public List<BookRecommendationForMeetingDto> BookRecommendations { get; set; } = [];

    public List<MeetingUserStateDto> UserStates { get; set; } = [];

    public void Mapping(Profile profile)
    {
        profile.CreateMap<Meeting, MeetingDto>()
            .ForMember(d => d.Status, opt => opt.MapFrom(s => s.Status.ToString()));
    }
}
